pyomo/gdp/plugins/binary_multiplication.py

```diff
@@ -0,0 +1,173 @@
+from .gdp_to_mip_transformation import GDP_to_MIP_Transformation
+from pyomo.common.config import ConfigDict, ConfigValue
+from pyomo.core.base import TransformationFactory
+from pyomo.core.util import target_list
+from pyomo.gdp import Disjunction
+from weakref import ref as weakref_ref
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+@TransformationFactory.register(
+    'gdp.binary_multiplication',
+    doc="Reformulate the GDP as an MINLP by multiplying f(x) <= 0 by y to get f(x) * y <= 0 where y is the binary corresponding to the Boolean indicator var of the Disjunct containing f(x) <= 0.",
+)
+class GDPBinaryMultiplicationTransformation(GDP_to_MIP_Transformation):
```

Comment:
Keeping in mind that I know very little about stuff in the `TransformationFactory`, is it standard for all methods to be private?

Reply:
Yeah, the only things that are going to get called in the outside world are `apply_to` and `create_using` and sometimes some methods that provide APIs for mapping between original and transformed components and vice versa. So at least with our current designs it's quite typical.

Reply:
_TIL_.